server/validate.py

```diff
@@ -0,0 +1,316 @@
+#!/usr/local/bin/python
+import argparse
+import functools
+import glob
+import logging
+import os
+import re
+import shutil
+import sys
+
+from pylode.profiles.ontpub import OntPub
+from rdflib import Graph, Literal, URIRef
+from rdflib.namespace import DCTERMS, OWL, RDF, RDFS
+
+_logger = logging.getLogger(__name__)
+
+ONTOLOGY_BASE = os.getenv('ONTOLOGY_BASE', os.path.join(os.getcwd()))
+
+def installation_directory(version: str) -> str | None:
+    """Retrieve the absolute installation directory of an ontology version.
+
+    :returns:
+        The installation path (OR SYMLINK PATH) if that version can be found,
+        Or None if there is no valid match.
+    """
+    expected = os.path.join(ONTOLOGY_BASE, version)
+    return expected if os.path.isdir(expected) else None
+
+
+def identify_version(value: str, fix_semver: bool = False) -> str:
+    """Extract and return an ontology version from a string that should contain one."""
+    match = re.search(r'v\d[\.\d]*', value)
+    version = match.group(0) if match else None
+    if version and fix_semver:
+        version_parts = version.split('.')
+        if len(version_parts) > 3:
+            # Some early releases had 4-part semver tags where the last part was a build-id.
+            # Those were NOT used in the ontology IRIs, so don't consider them part of the version.
+            version_parts = version_parts[:-1]
+        if len(version_parts) > 2 and version_parts[-1] == 0:
```

Comment:
`version_parts` will always be a list of str, and `"0" != 0`.
```suggestion
        if len(version_parts) > 2 and version_parts[-1] == "0":
```

Reply:
Also fixed!